prow/config/config.go

```diff
@@ -484,11 +484,20 @@ type Plank struct {
 	// PodUnscheduledTimeout is after how long the controller will abort a prowjob
 	// stuck in an unscheduled state. Defaults to one day.
 	PodUnscheduledTimeout *metav1.Duration `json:"pod_unscheduled_timeout,omitempty"`
+
 	// DefaultDecorationConfigs holds the default decoration config for specific values.
-	// This config will be used on each Presubmit and Postsubmit's corresponding org/repo, and on Periodics
-	// if extraRefs[0] exists.
-	// Use `org/repo`, `org` or `*` as a key.
-	DefaultDecorationConfigs map[string]*prowapi.DecorationConfig `json:"default_decoration_configs,omitempty"`
+	// Each entry in the slice specifies Repo and Cluster regexp filter fields to
+	// match against jobs and a corresponding DecorationConfig. All entries that
+	// match a job are used. Later matching entries override the fields of earlier
+	// matching entries.
+	// Alternatively this field may be type `map[string]*prowapi.DecorationConfig`
+	// Use `org/repo`, `org` or `*` as a key to match against jobs. (Periodics
+	// use extra_refs[0] for matching if present.)
+	DefaultDecorationConfigs []*DefaultDecorationConfigEntry `json:"-"`
```

Comment:
The godocs detail the types that are accepted. I wanted to use a single field to avoid allowing multiple sources of truth, but we could enforce mutual exclusivity with additional validation if that is preferred. I don't think having separate top level config fields for each type will make the code simpler to read, but I don't feel strongly about it and am happy to change it if preferred.

Reply:
Well right now its not enough to look at the types, you need to look at the loading code to understand the possible formats. Having two mutually exclusive fields would restore the ability to just look at the type to understand the format, so I would argue its simpler or WDYT?

Reply:
FWIW you don't need to look at the loading code, the accepted types are included in the godoc. Still happy to change if you prefer, but now I think this might be problematic for your reserialization requirements so please confirm: https://github.com/kubernetes/test-infra/pull/20656#issuecomment-785428469